ui/components/app/auto-detect-nft/auto-detect-nft-modal.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useCallback, useContext } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalHeader,
+  Box,
+  Text,
+  ModalBody,
+  ModalFooter,
+} from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  AlignItems,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextAlign,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { setUseNftDetection } from '../../../store/actions';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import { getProviderConfig } from '../../../ducks/metamask/metamask';
+import { getCurrentLocale } from '../../../ducks/locale/locale';
+import { ORIGIN_METAMASK } from '../../../../shared/constants/app';
+
+type AutoDetectNftModalProps = {
+  isOpen: boolean;
+  onClose: (arg: boolean) => void;
+};
+function AutoDetectNftModal({ isOpen, onClose }: AutoDetectNftModalProps) {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+  const { chainId } = useSelector(getProviderConfig);
+
+  const handleNftAutoDetection = useCallback(
+    (val) => {
+      trackEvent({
+        event: val
+          ? MetaMetricsEventName.NftAutoDetectionEnableModal
+          : MetaMetricsEventName.NftAutoDetectionDisableModal,
+        category: MetaMetricsEventCategory.Navigation,
+        properties: {
+          chain_id: chainId,
+          locale: getCurrentLocale,
+          referrer: ORIGIN_METAMASK,
+        },
+      });
+      dispatch(setUseNftDetection(val));
```

Comment:
Something I notice.

When I manually click the setting to enable nft detection, it also flips on the `Display NFT media`, because the settings page has logic to do so:

https://github.com/MetaMask/metamask-extension/assets/3500406/41b8fc76-c9d7-49df-a2f8-8b148b0d4c66

But when I enable nft detection through this modal, it does not enable `Display NFT media`.  I think it probably should:


https://github.com/MetaMask/metamask-extension/assets/3500406/73d74645-e9c0-451f-909c-35ff07676f60

Reply:
That is right, i will delay this fix on the modal a bit just to check if im removing it in favor of the updated banner notice.
And if we end up only keeping the banner notice, i agree that clicking on the "allow" should also enable the display NFT media

Reply:
Fixed this on the nft modal 🙏